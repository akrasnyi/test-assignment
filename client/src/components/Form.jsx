import React from 'react'
import Button from '@material-ui/core/Button'
import TextField from '@material-ui/core/TextField'
import Snackbar from '@material-ui/core/Snackbar'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const StyledForm = styled.form`
  padding: 20px;

  .spacing {
    margin: 10px 0;
  }
`

const Form = props => {
  const { handleInputChange, handleValidation, messageToggle, isMessageOpen } = props
  let isFnameValid = props.validation.fname
  let isLnameValid = props.validation.lname
  let isEmailValid = props.validation.email
  let isIbanValid = props.validation.iban
  return (
    <StyledForm onSubmit={props.handleSubmit}>
      <TextField
        id='fname'
        name='fname'
        className='spacing'
        label='First Name'
        fullWidth
        variant='outlined'
        error={!isFnameValid}
        helperText={!isFnameValid ? 'First Name is required' : ''}
        onChange={handleInputChange}
        onBlur={handleValidation}
      />
      <TextField
        id='lname'
        name='lname'
        className='spacing'
        label='Last Name'
        fullWidth
        variant='outlined'
        error={!isLnameValid}
        helperText={!isLnameValid ? 'Last Name is required' : ''}
        onChange={handleInputChange}
        onBlur={handleValidation}
      />
      <TextField
        id='email'
        name='email'
        className='spacing'
        label='Email'
        fullWidth
        type='email'
        variant='outlined'
        error={!isEmailValid}
        helperText={!isEmailValid ? 'Value should be a valid email' : ''}
        onChange={handleInputChange}
        onBlur={handleValidation}
      />
      <TextField
        id='iban'
        name='iban'
        className='spacing'
        label='IBAN'
        fullWidth
        variant='outlined'
        defaultValue='UA903052992990004149123456789'
        error={!isIbanValid}
        helperText={!isIbanValid ? 'IBAN should be a valid' : ''}
        onChange={handleInputChange}
        onBlur={handleValidation}
      />
      <Button type='submit' fullWidth variant='outlined' color='primary'>
        Submit
      </Button>
      <Snackbar
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'center'
        }}
        variant='success'
        open={isMessageOpen}
        onClose={messageToggle}
        autoHideDuration={3000} // 3 seconds close timeout
        message={
          <span id='message-id'>
            {isFnameValid && isLnameValid && isEmailValid && props.validation.isTouched
              ? 'Congratz! All data is valid'
              : 'Fix problems with input data'}
          </span>
        }
      />
    </StyledForm>
  )
}

Form.propTypes = {
  validation: PropTypes.object,
  handleInputChange: PropTypes.func,
  handleValidation: PropTypes.func,
  messageToggle: PropTypes.func,
  isMessageOpen: PropTypes.bool
}

export default Form
