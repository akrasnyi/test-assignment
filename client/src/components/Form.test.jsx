import React from 'react'
import Form from './Form'
import renderer from 'react-test-renderer'

const validation = {
  fname: true,
  lname: true,
  email: true,
  iban: true,
  isTouched: false
}

it('renders correctly', () => {
  const tree = renderer.create(<Form validation={validation} />).toJSON()
  expect(tree).toMatchSnapshot()
})
