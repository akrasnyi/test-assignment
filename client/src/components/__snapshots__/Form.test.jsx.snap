// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="sc-bdVaJa gpAajL"
>
  <div
    className="MuiFormControl-root MuiTextField-root spacing MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="fname"
    >
      First Name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-58 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-59"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="fname"
        name="fname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
    
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root spacing MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="lname"
    >
      Last Name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-58 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-59"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="lname"
        name="lname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
    
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root spacing MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="email"
    >
      Email
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-58 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-59"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="email"
      />
    </div>
    
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root spacing MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="iban"
    >
      IBAN
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-58 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-59"
          style={
            Object {
              "width": 0,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        defaultValue="UA903052992990004149123456789"
        disabled={false}
        id="iban"
        name="iban"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
    
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Submit
    </span>
  </button>
</form>
`;
